### DISCUSSION TITLE
Event System Tracking

### DISCUSSION THREAD

**Discussion by grenade44** [RetroAchievements/RAWeb#665] [Ideas]
The RA events have been great and a nice system that automatically does everything from awarding points to awarding titles to updating the leader boards would be great addition to the RA site.

**Comment by meleu** [comment]
@grenade44 could you elaborate this request? Is it related with the "audit log" thing? Like suggested here in #11.

**Comment by grenade44** [comment]
I don’t think its the same. 

When RA run an event, someone has to do all the work and manually adding points to the leaderboard when users unlock achievements in certain games. 

The system could update the leaderboard when users unlock achievement and award the titles (Halloween event was top title was pumpkin king) on the gamers page and game award for playing the event.

Also other stuff like bonus point and developer points for genre of set released while the event is running could be added.

This would make running event a lot easier.

**Comment by luchaos** [comment]
as for events and achievements, badges/awards related to them - i'd envision that as something that will be run independently in the future of what is currently games/achievements as it's a more RA community domain specific thing. 
meaning the existing event "games", their achievements and awards would be migrated to their own space.
that would make designing features for events much easier and flexbile as well as they wouldn't have to follow standard achievement procedures/triggers - a separate set of roles and permission can be established this way as well. 
this prevents introducing required metadata for games and achievements that are not related to "actual" games, keeping the datasets clean. instead, events could have other resources attached (like leaderboards) and/or triggers set up to assign points when x/specific achievements were unlocked in specified games.

**Comment by meleu** [comment]
@grenade44 Understood. Thanks! ;)

**Comment by Jamiras** [comment]
Expanding on this idea, I proposed the following for the AotW 2019 challenge:
* Create a "game" entry for the event (similar to http://retroachievements.org/game/13755)
* Instead of populating with six dummy achievements (minimum for a badge), it should be populated with 52 dummy achievements - one for each week that the event runs.
  -  pre-populating all 52 achievements would prevent awarding a "completion" badge until all 52 weeks were completed.
  - after an achievement is chosen for each week, the next dummy achievement would be updated to use the name and icon of the chosen achievement, as well as the game and system. i.e. the first achievement would have been "Week 1: Sleep Well (Little Nemo: The Dream Master) [NES]", for http://retroachievements.org/achievement/7091
  - it would be worth 1 point (automatically doubled to 2 for hardcore)
  - it would still have to be manually awarded, until some form of official event integration is provided, but would use the database for long-term tracking instead of relying on external spreadsheets or forum posts.
* The "High Scores" list would automatically track users who have completed the most AotWs overall (based on total points earned within the event) as the achievements would actually award the points.
* Players could view the event page to see:
  - which AotWs were chosen for each week
  - which they've completed
  - completion ratios for all AotWs so far: "won by 19 (16) of 26" 
  - and overall completion via the Achievement Distribution graph.
* At the end of the year, if we wanted to provide a participation badge or a badge for achieving some percentage of the AotWs, we would need new functionality. We currently do not have any way to award a game badge without awarding all of the achievements for the game.

Initial support was very much for doing something like this even though it was an almost entirely manual process, but there was concerns that it would affect player's Completion%. Would it be difficult to exclude games for the "event" console from that calculation?

**Reply by Jamiras** [reply]
This proposal has been implemented, and is currently being used for AotW 2025.

**Reply by Hexadigital** [reply]
Thank you for the implementation! I love the new event page and I'm definitely interested in seeing how the stats on it evolve throughout AotW 2025.

Does it track softcore player progress as well? I understand that AotW 2025 doesn't currently offer event badges for softcore players, but I was wondering if I could use the page to join the event in spirit (and maybe get a badge retroactively if that policy ever changes).

**Comment by kdecks** [comment]
This is a great way to handle it.

**Comment by Ben-Ghazi** [comment]
> At the end of the year, if we wanted to provide a participation badge or a badge for achieving some percentage of the AotWs, we would need new functionality. We currently do not have any way to award a game badge without awarding all of the achievements for the game.

Actually we kind of do, when I was messing around with the API I found a way to give myself the Patreon badge with https://retroachievements.org/requestupdateuser.php?t=[Username]&p=2&v=1 we could possibly expand upon that as a way to award event badges without having to create a game entry.
